regolith/export.go

```diff
@@ -12,98 +12,145 @@ import (
+	if exportTarget.WorldPath != "" {
+		if exportTarget.WorldName != "" {
+			return "", "", burrito.WrappedError(
+				"Using both \"worldName\" and \"worldPath\" is not" +
+					" allowed.")
+		}
+		wPath, err := ResolvePath(exportTarget.WorldPath)
 		if err != nil {
 			return "", "", burrito.WrapError(
-				err, "Failed to resolve behavior pack path.")
+				err, "Failed to resolve world path.")
 		}
-		rpPath, err = ResolvePath(exportTarget.RpPath)
+		bpPath = filepath.Join(
+			wPath, "behavior_packs", bpName)
+		rpPath = filepath.Join(
+			wPath, "resource_packs", rpName)
+	} else if exportTarget.WorldName != "" {
+		dir, err := FindMojangDir()
 		if err != nil {
 			return "", "", burrito.WrapError(
-				err, "Failed to resolve resource pack path.")
+				err, "Failed to find \"com.mojang\" directory.")
 		}
-	} else if exportTarget.Target == "world" {
-		if exportTarget.WorldPath != "" {
-			if exportTarget.WorldName != "" {
-				return "", "", burrito.WrappedError(
-					"Using both \"worldName\" and \"worldPath\" is not" +
-						" allowed.")
-			}
-			wPath, err := ResolvePath(exportTarget.WorldPath)
-			if err != nil {
-				return "", "", burrito.WrapError(
-					err, "Failed to resolve world path.")
-			}
-			bpPath = filepath.Join(
-				wPath, "behavior_packs", name+"_bp")
-			rpPath = filepath.Join(
-				wPath, "resource_packs", name+"_rp")
-		} else if exportTarget.WorldName != "" {
-			dir, err := FindMojangDir()
-			if err != nil {
-				return "", "", burrito.WrapError(
-					err, "Failed to find \"com.mojang\" directory.")
-			}
-			worlds, err := ListWorlds(dir)
-			if err != nil {
-				return "", "", burrito.WrapError(err, "Failed to list worlds.")
-			}
-			for _, world := range worlds {
-				if world.Name == exportTarget.WorldName {
-					bpPath = filepath.Join(
-						world.Path, "behavior_packs", name+"_bp")
-					rpPath = filepath.Join(
-						world.Path, "resource_packs", name+"_rp")
-				}
+		worlds, err := ListWorlds(dir)
+		if err != nil {
+			return "", "", burrito.WrapError(err, "Failed to list worlds.")
+		}
+		for _, world := range worlds {
+			if world.Name == exportTarget.WorldName {
+				bpPath = filepath.Join(
+					world.Path, "behavior_packs", bpName)
+				rpPath = filepath.Join(
+					world.Path, "resource_packs", rpName)
 			}
-		} else {
-			err = burrito.WrappedError(
-				"The \"world\" export target requires either a " +
-					"\"worldName\" or \"worldPath\" property")
 		}
-	} else if exportTarget.Target == "local" {
-		bpPath = "build/BP/"
-		rpPath = "build/RP/"
 	} else {
-		err = burrito.WrappedErrorf(
-			"Export target %q is not valid", exportTarget.Target)
+		err = burrito.WrappedError(
+			"The \"world\" export target requires either a " +
+				"\"worldName\" or \"worldPath\" property")
+	}
+	return
+}
+
+func GetExportNames(exportTarget ExportTarget, ctx RunContext) (bpName string, rpName string, err error) {
```

Comment:
The benefit is the flexibility. You can choose whether is a prefix or a suffix or maybe they want to format the name etc

Reply:
I was writing the documentation for this and looked into the code a little bit more and honestly I don't see when this would be useful. The set of exposed variables is weird and one of them is very confusing to me. I'm looking at this part of the code:

https://github.com/Bedrock-OSS/regolith/blob/de85f24185e178a923c5399d4d264e4e052d423f/regolith/evaluator.go#L38-L57

- "name" - this is probably the only useful variable on the list. That said, the benefit is tiny. Using it means that if you want to export to a directory named after the project, you don't have to change the value twice when you rename the project.
- "author" - maybe someone would like to export to a directory that includes their name.
- "os" - probably useless
- "arch" - probably useless
- "debug" - I guess someone might want to export to different path when they're debugging. But in what case?
  - if you use `development` export it would most likely create multiple packs with the same UUID in your dev packs folder. This could potentially cause problems in Minecraft.
  - in my experience `local` export target is used only for Regolith development and to showing people examples, but if someone is using some custom workflow where `local` is useful then they are probably using some other tool to export the files to the actual location and they probably have that tool set up in such a way that it expects specific export path.
  - `exact` doesn't support this feature
  - `world` - same problem as with `development`
  - `preview` - same problem as with `development` and `world`
- "version" - useless. Do you run multiple versions of Regolith?
- "profile" - similar benefit to "name" but probably even smaller.
- "filteLocation" - honestly I don't even know what that is.
- "settings" - same here. I don't know what it does. I think it normally would have filter settings but in context of creating a name of an export directory for a profile there is no single filter that could be used to provide data for this.
- "project" contains the data I discussed in first 2 points.

If you find this feature useful for your specific workflow that's fine. I'm not saying that we have to remove it. TBH it doesn't affect me at all because I use `exact` export target, but I don't know what to write in the documentation.

I guess I could mention only the "author", "debug", "name" and "profile" properties. I could also add descriptions for "os" and "arch". The rest of the variables is either useless or I don't understand what they do (which means I can't really document them).

Reply:
Updated documentation: https://github.com/Bedrock-OSS/regolith/pull/271/commits/23c269695fcfa2dad802d8690bc90c3af7c7c5c3